hopsworks-common/src/main/java/io/hops/hopsworks/common/hive/HiveScratchdirCleaner.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2019, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ *
+*/
+
+package io.hops.hopsworks.common.hive;
+
+import io.hops.hopsworks.common.dao.hdfs.inode.Inode;
+import io.hops.hopsworks.common.dao.hdfs.inode.InodeFacade;
+import io.hops.hopsworks.common.hdfs.DistributedFileSystemOps;
+import io.hops.hopsworks.common.hdfs.DistributedFsService;
+import io.hops.hopsworks.common.util.Settings;
+import io.hops.hopsworks.common.yarn.YarnClientService;
+import io.hops.hopsworks.common.yarn.YarnClientWrapper;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.yarn.api.records.ApplicationReport;
+import org.apache.hadoop.yarn.api.records.QueueInfo;
+import org.apache.hadoop.yarn.api.records.YarnApplicationState;
+import org.apache.hadoop.yarn.client.api.YarnClient;
+import org.apache.hadoop.yarn.exceptions.YarnException;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import javax.ejb.DependsOn;
+import javax.ejb.EJB;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.ejb.Timeout;
+import javax.ejb.TimerService;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Singleton
+@Startup
+@DependsOn("Settings")
+public class HiveScratchdirCleaner {
+
+  @EJB
+  private Settings settings;
+  @EJB
+  private InodeFacade inodeFacade;
+  @EJB
+  private DistributedFsService dfs;
+  @EJB
+  private YarnClientService yarnService;
+  @Resource
+  private TimerService timerService;
+
+  private static final Logger logger = Logger.getLogger(HiveScratchdirCleaner.class.getName());
+
+  private Set<String> applicationTypeSet = null;
+  private EnumSet<YarnApplicationState> applicationStateEnumSet = null;
+
+  @PostConstruct
+  private void init() {
+    // SparkSQL creates a scratchdir as well
+    applicationTypeSet = new HashSet<>(Arrays.asList("TEZ", "SPARK"));
+
+    applicationStateEnumSet = EnumSet.of(YarnApplicationState.NEW, YarnApplicationState.GENERATING_SECURITY_MATERIAL,
```

Comment:
@SirOibaf can we reuse the JobState.getRunningStates here? Cause it basically has the same semantics. Or we can have a runningAppStates EnumSet in the `YarnApplicationStateFacade`

Reply:
No, we can't use the `JobState` as the EnumSet is fed directly into the `YarnClient` which understands only `YarnApplicationState` and not `JobState`. We will have to convert all the `JobStates` back to `YarnApplicationStates` - which I think it's out of the scope of the `JobState` enum.

I can move the definition to the `YarnApplicationStateFacade` in case we'll need it again in the future.